src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -649,6 +649,38 @@ public function dumpFile($filename, $content)
     }
 
     /**
+     * Atomically appends content to an existing file.
+     *
+     * @param string $filename The file for which to append content
+     * @param string $content  The content to append
+     *
+     * @throws IOException If the file is not writable
+     */
+    public function appendToFile($filename, $content)
+    {
+        $dir = dirname($filename);
+
+        if (!is_dir($dir)) {
+            $this->mkdir($dir);
+        } elseif (!is_writable($dir)) {
+            throw new IOException(sprintf('Unable to write to the "%s" directory.', $dir), 0, null, $dir);
+        }
+
+        $tmpFile = $this->tempnam(dirname($filename), basename($filename));
+
+        if (false === $originalContent = @file_get_contents($filename)) {
```

Comment:
What about forwarding to `dumpFile` instead in this case? So above checks can be removed..